### PR TITLE
Update TechDocs For Models

### PR DESCRIPTION
**Pull request by Jdubrick**
This PR updates the TechDocs for the 2 models that are referenced so far as part of our catalog. These TechDocs can be used as a template for anyone wishing to utilize TechDocs for their catalogs. 

This is part of https://issues.redhat.com/browse/DEVAI-148

Documentation: https://docs.google.com/document/d/193uwS-uji9LTh4Npax5ABkHZnZzkLns9sN0c2ZxNZwc/edit?usp=sharing

You can test these changes by importing the catalog into your RHDH instance. Setup instructions can be found [here](https://github.com/redhat-ai-dev/model-catalog/blob/main/readme.md) and there are also example screenshots in the Google document.

### CODE REVIEW

**Comment by Jdubrick** [comment]
I don't believe reviewers were set up for this repository yet so I am just going to tag the team @redhat-ai-dev/ai-development-team

File: ibm-granite-8b-code-instruct/docs/ethics.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Ethical Considerations
+
+Granite Code Instruct models are primarily finetuned using instruction-response pairs across a specific set of programming languages. Thus, their performance may be limited with out-of-domain programming languages. In this situation, it is beneficial providing few-shot examples to steer the model's output. Moreover, developers should perform safety testing and target-specific tuning before deploying these models on critical applications. The model also inherits ethical considerations and limitations from its base model. For more information, please refer to [Granite-8B-Code-Base-4K model card](https://huggingface.co/ibm-granite/granite-8b-code-base-4k).
```
Comment (thepetk) [inline ibm-granite-8b-code-instruct/docs/ethics.md]:
```suggestion
Granite Code Instruct models are primarily fine-tuned using instruction-response pairs across a specific set of programming languages. Thus, their performance may be limited with out-of-domain programming languages. In this situation, it is beneficial providing few-shot examples to steer the model's output. Moreover, developers should perform safety testing and target-specific tuning before deploying these models on critical applications. The model also inherits ethical considerations and limitations from its base model. For more information, please refer to [Granite-8B-Code-Base-4K model card](https://huggingface.co/ibm-granite/granite-8b-code-base-4k).
```

File: ibm-granite-8b-code-instruct/docs/model.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# IBM Granite 8B Code-Instruct
+
+## Model Description
+
+Granite-8B-Code-Instruct-4K is a 8B parameter model fine tuned from Granite-8B-Code-Base-4K on a combination of **permissively licensed** instruction data to enhance instruction following capabilities including logical reasoning and problem-solving skills.
```
Comment (thepetk) [inline ibm-granite-8b-code-instruct/docs/model.md]:
```suggestion
Granite-8B-Code-Instruct-4K is a 8B parameter model fine-tuned from Granite-8B-Code-Base-4K on a combination of **permissively licensed** instruction data to enhance instruction following capabilities including logical reasoning and problem-solving skills.
```

**Review by thepetk — COMMENTED** [review]
I just left a minor suggestion for fine-tuning

**Comment by thepetk** [comment]
I've also tested it on a cluster of and it works fine:

* I can list resources:
Screenshot from 2024-09-24 13-06-46

* I can access docs:
Screenshot from 2024-09-24 13-07-49

**Review by johnmcollier — COMMENTED** [review]
Overall looks really good @Jdubrick, I think this modular approach is exactly what we want for the techdocs.

Only bit of feedback I would have is maybe include a small comment in markdown (so it doesn't show up when rendered) explaining how the information on each page was fetched? Might be useful when we reference this externally

**Comment by Jdubrick** [comment]
> Overall looks really good @Jdubrick, I think this modular approach is exactly what we want for the techdocs.
> 
> Only bit of feedback I would have is maybe include a small comment in markdown (so it doesn't show up when rendered) explaining how the information on each page was fetched? Might be useful when we reference this externally

@johnmcollier I added a blurb regarding each section and the sourced information as part of this commit: https://github.com/redhat-ai-dev/model-catalog/pull/1/commits/5fb2aabb4fb753bf5dd35b8dab5a4b48904e3c07

**Review by johnmcollier — APPROVED** [review]
LGTM

**Review by thepetk — APPROVED** [review]
lgtm

**Pull request merged**